support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -417,9 +425,16 @@ AccessControlled accessControlled() {
                         cookie = UUID.randomUUID().toString();
                         // Switches the label to a self-label, so if the executable is killed and restarted via ExecutorPickle, it will run on the same node:
                         label = computer.getName();
+
                         EnvVars env = computer.getEnvironment();
                         env.overrideAll(computer.buildEnvironment(listener));
                         env.put(COOKIE_VAR, cookie);
+                        if (exec.getOwner() instanceof MasterComputer) {
+                            env.put("NODE_NAME", "master");
+                        } else {
+                            env.put("NODE_NAME", label);
+                        }
+
```

Comment:
It is fine to defer handling of `NODE_LABELS` and `EXECUTOR_NUMBER` to a future change, but you should leave here a `TODO` comment noting that you are doing so, and mentioning where these come from normally (`CoreEnvironmentContributor` IIRC).